secrethub/provider.go

```diff
@@ -60,6 +60,12 @@ func configureProvider(d *schema.ResourceData) (interface{}, error) {
 		return nil, err
 	}
 
+	credential := client.DefaultCredential()
+	_, err = credential.Read()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is there a nicer way to check whether the client is authenticated?

Reply:
The client should be able to determine itself whether or not authentication is required and return an error accordingly if needed. I believe it shouldn't be necessary to handle this separately here.

Why did you add it? Maybe this is an indication we should change something in the SDK.

If you prefer, we can also have a quick call about this, might be easier.